Tests/ObjCTests.mm

```diff
@@ -62,7 +62,9 @@ static void checkIt(id obj, const char* json) {
     checkIt(@123456789,     "123456789");
     checkIt(@123456789012,  "123456789012");
     checkIt(@(INT64_MAX),   "9223372036854775807");
-    checkIt(@(UINT64_MAX),  "18446744073709551615");
+    // It was found that @() may not turn UINT64_MAX to unsigned. In the following,
+    // we apply an explicit conversion
+    checkIt(@([@UINT64_MAX unsignedLongLongValue]), "18446744073709551615");
```

Comment:
This is unnecessarily convoluted. You create an NSNumber, turn it back into a native number, then create another NSNumber from that.
It would be much simpler to write `[NSNumber numberWithUnsignedLongLong: UINT64_MAX]`.
Or else `@((unsigned long long)UINT64_MAX)`

Reply:
Thanks for the advice. This is test code and won't be in the product. I will change when ported the the master branch.